### PR TITLE
Enhance delete on cvmfs_server ingest

### PR DESCRIPTION
**Pull request by siscia**
This PR allow for deleting entities without ingesting anything.

I also made more rigorous analysis on what parameters need to be checked in order to abort the command for lacking of parameters. (First time I used Karnaugh map :)  )

### CODE REVIEW

File: cvmfs/sync_union_tarball.cc [hunk 1]
```diff
@@ -100,6 +107,9 @@ void SyncUnionTarball::Traverse() {
     }
   }
 
+  /* we are simplying deleting entity from  the repo*/
```
Comment (jblomer) [inline cvmfs/sync_union_tarball.cc]: Nit: please use `//` style comments (in order to allow quickly commenting larger blocks with `/* */` if needed)

File: test/src/657-tarball_remove_base_directory/main [hunk 2]
```diff
@@ -138,6 +138,37 @@ cvmfs_run_test() {
...
+  file1=$to_keep_dir/tarball_foo/a/b/c/1.txt
+  file2=$to_keep_dir/tarball_foo/a/b/c/2.txt
+  cvmfs_server ingest -d $file1 -d $file2 $CVMFS_TEST_REPO || return $?
+  if [ -f $file1 ] || [ -f $file2 ]; then
+    echo "*** Error found file $file1 or $file2"
+    return 11
+  else
+    echo "*** Removed files: $file1 and $file2"
+  fi
+
```
Comment (jblomer) [inline test/src/657-tarball_remove_base_directory/main]: It would be good to add `cvmfs_server check -i` commands after the various ingestion steps.

Comment (siscia) [reply test/src/657-tarball_remove_base_directory/main]: Done!

**Comment by siscia** [comment]
This one seems to build ok:

https://epsft-jenkins.cern.ch/view/CVMFS/job/CvmfsIncrementalBuildOnPR/54/parameters/

**Pull request merged**